### PR TITLE
refactor: move dependency validations to domain pkg

### PR DESCRIPTION
**Pull request by kentakozuka**
1. Moves dependency validation/sort functions to domains.
2. Use the moved functions for validation in API layer.

This is also a part of https://github.com/bucketeer-io/bucketeer/issues/904.
UpdateFeature API will use this validation.

### CODE REVIEW

File: pkg/feature/api/feature.go [hunk 1]
```diff
@@ -1162,9 +1165,39 @@ func (s *FeatureService) ArchiveFeature(
...
 		return nil, err
 	}
-	if err := validateArchiveFeatureRequest(req, features, localizer); err != nil {
-		return nil, err
+	var tgtF *domain.Feature
+	for _, f := range features {
+		if f.Id == req.Id {
+			tgtF = &domain.Feature{Feature: f}
+			break
+		}
```
Comment (cre8ivejp) [inline pkg/feature/api/feature.go:1173]: If the request ID is not in the index zero, it will return an internal.

Comment (kentakozuka) [reply pkg/feature/api/feature.go:1173]: Thanks! fixed [4a00e23](https://github.com/bucketeer-io/bucketeer/pull/1076/commits/4a00e2383fe2ffba0e4e1033c3d485ed6439873d)

**Comment by kentakozuka** [comment]
@cre8ivejp Fixed 👍

**Review by cre8ivejp — APPROVED** [review]
👍

**Pull request merged**